### ISSUE TITLE
[BUG] TextBuffer.selection_position is causing segmentation fault

### ISSUE THREAD

**Issue by mvasi90** [fltk-rs/fltk-rs#834]
Hello,

The method `buffer.selection_position()` is causing segmentation fault.

## To Reproduce
```rust
use fltk::{app, button::Button, frame::Frame, group::{Pack, PackType}, prelude::{DisplayExt, GroupExt, WidgetBase, WidgetExt}, text::{TextBuffer, TextEditor}, window::Window};

fn main() {
    let app = app::App::default();
    let mut w = Window::default().with_size(400, 200).with_label("Highlight");

    let mut vpack = Pack::new(4, 4, 392, 163, "");
    vpack.set_spacing(4);

    let mut editor = TextEditor::default().with_size(392, 163);

    let mut hpack = Pack::default().with_size(392, 25).with_type(PackType::Horizontal);
    hpack.set_spacing(4);
    let mut frame = Frame::default().with_size(196, 25);
    let mut btn = Button::default().with_size(196, 25).with_label("Get selection range");

    hpack.end();
    vpack.end();

    w.end();
    w.show();

    let buf = TextBuffer::default();
    editor.set_buffer(buf);

    btn.set_callback(move |_| {
        if let Some(mut buff) = editor.buffer() {
            match buff.selection_position() {
                Some((start, end)) => {
                    frame.set_label(format!("{}:{}", start, end).as_str());
                },
                None => {
                    frame.set_label("No selection");
                }
            }
        }
    });

    app.run().unwrap();
}
```

## Desktop info
 - ArchLinux distribution. Sway WM with Wayland

 - `cargo build -vv`
       Fresh cc v1.0.69
       Fresh unicode-xid v0.2.2
       Fresh lazy_static v1.4.0
       Fresh cmake v0.1.45
       Fresh proc-macro2 v1.0.28
       Fresh libc v0.2.98
       Fresh quote v1.0.9
       Fresh bitflags v1.2.1
       Fresh syn v1.0.74
       Fresh raw-window-handle v0.3.3
       Fresh fltk-sys v1.1.6
       Fresh fltk-derive v1.1.5
       Fresh fltk v1.1.6
 - `rustup toolchain list
       stable-x86_64-unknown-linux-gnu (default)

**Comment by MoAlyousef** [comment]
Hello
Thanks for the report.
I can confirm. I've pushed a fix to master and will publish later in the evening.

**Comment by MoAlyousef** [comment]
I've published version 1.1.7 with the fix

**Comment by mvasi90** [comment]
Wow, very fast!
Confirmed, it works.

Thanks.
